activerecord/test/cases/relation/where_test.rb

```diff
@@ -114,6 +115,25 @@ def test_polymorphic_nested_array_where
       assert_equal expected.to_sql, actual.to_sql
     end
 
+    def test_polymorphic_array_where_multiple_types
+      treasure_1 = Treasure.new
+      treasure_1.id = 1
+
+      treasure_2 = Treasure.new
+      treasure_2.id = 2
+
+      car = Car.new
+      car.id = 3
+
+      table = PriceEstimate.arel_table
+      where_arel = table.grouping(table[:estimate_of_type].eq('Treasure').and(table[:estimate_of_id].in([1, 2])))
+                        .or(table.grouping(table[:estimate_of_type].eq('Car').and(table[:estimate_of_id].eq(3))))
+      expected = table.project(table[Arel.star]).where(where_arel)
+      actual   = PriceEstimate.where(estimate_of: [treasure_1, treasure_2, car])
+
+      assert_equal expected.to_sql, actual.to_sql
```

Comment:
Ahh- the other tests in this file are asserting based of generated sql, so I was just copying that format.  I can swap to checking the query result, but just wanted to double check as that would slow down the test (since it's hitting the db now) and may add complexity, as I'll need to load the fixtures for these models or at least my test will need to know how to create valid and savable models.